MantidQt/CustomInterfaces/test/EnggDiffractionPresenterTest.h

```diff
@@ -1190,6 +1191,182 @@ class EnggDiffractionPresenterTest : public CxxTest::TestSuite {
     pres.notify(IEnggDiffractionPresenter::FitPeaks);
   }
 
+  // shahroz Fitting test begin here
+  void test_fitting_runno_valid_single_run() {
+    testing::NiceMock<MockEnggDiffractionView> mockView;
+    EnggDiffPresenterNoThread pres(&mockView);
+
+    // inputs from user
+    EXPECT_CALL(mockView, getFittingRunNo())
+        .Times(2)
+        .WillRepeatedly(Return(std::string(g_focusedBankFile)));
+
+    EXPECT_CALL(mockView, getFittingRunNumVec())
+        .Times(1)
+        .WillOnce(Return(m_ex_run_number));
+
+    EXPECT_CALL(mockView, splitFittingDirectory(testing::_))
+        .Times(1)
+        .WillOnce(Return(m_ex_run_number));
+
+    // No errors/0 warnings. There will be no errors or warnings
+    EXPECT_CALL(mockView, userError(testing::_, testing::_)).Times(0);
+    EXPECT_CALL(mockView, userWarning(testing::_, testing::_)).Times(1);
+
+    pres.notify(IEnggDiffractionPresenter::FittingRunNo);
+  }
```

Comment:
I'd add here the `VerifyAndClearExpectations()` as it has been added for the other tests, after sending the notification. To make sure that the warnings don't remain hidden.